experimental/yarpl/include/yarpl/Flowables.h

```diff
@@ -108,6 +108,25 @@ class Flowables {
     return Flowable<T>::create(std::move(lambda));
   }
 
+  template <typename T, typename TGenerator>
+  static Reference<Flowable<T>> fromGenerator(TGenerator generator) {
+    auto lambda = [generator = std::move(generator)]
+    (Subscriber<T>& subscriber, int64_t requested) {
+      int64_t generated = 0;
+      try {
+        while (generated < requested) {
+          subscriber.onNext(generator());
```

Comment:
I think we should leave this as is.  The spec has no guarantees about synchronizing `onNext()` with `cancel()`, and I don't think there's a great solution that fits all client use-cases.